core/src/main/java/org/apache/brooklyn/core/internal/BrooklynInitialization.java

```diff
@@ -61,21 +61,22 @@ public static void initLegacyLanguageExtensions() {
      * 
      */
     
-    public synchronized static void initAll() {
-        if (done.get()) return;
-        initTypeCoercionStandardAdapters();
-        initSecureKeysBouncyCastleProvider();
-        initNetworking();
-        initPortRanges();
-        initLegacyLanguageExtensions();
-        done.set(true);
+    public static void initAll() {
+        if (done.compareAndSet(false, true)) {
+            initTypeCoercionStandardAdapters();
+            initSecureKeysBouncyCastleProvider();
+            initNetworking();
+            initPortRanges();
+            initLegacyLanguageExtensions();
+        }
     }
 
     @SuppressWarnings("deprecation")
     public synchronized static void reinitAll() {
-        done.set(false);
-        org.apache.brooklyn.util.core.BrooklynLanguageExtensions.reinit();
-        initAll();
+        if (done.compareAndSet(true, false)) {
```

Comment:
This changes the behaviour - won't run if `initAll` is not called before that. Better stick to the original, wdyt?